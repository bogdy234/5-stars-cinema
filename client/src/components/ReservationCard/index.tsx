import { FC, ReactElement } from "react";
import CONSTANTS from "../../constants";
import { SeatPosition } from "../../interfaces/seat";
import { Ticket } from "../../interfaces/ticket";

interface ReservationCardProps {
  reservationId: string;
  hallNumber: number;
  movie: string;
  time: string;
  seats: SeatPosition[];
  typesOfTickets: Ticket;
  totalPrice: number;
}

const {
  RESERVATION,
  RESERVATION_ID,
  SEATS,
  HALL,
  NORMAL_TICKETS,
  REDUCED_TICKETS,
  TOTAL_PRICE,
} = CONSTANTS.TEXT.RESERVATION_CARD;

const ReservationCard: FC<ReservationCardProps> = ({
  reservationId,
  hallNumber,
  movie,
  time,
  seats,
  typesOfTickets,
  totalPrice,
}): ReactElement => {
  return (
    <div className="w-full">
      <div className="w-[18rem] sm:w-[25rem] md:w-[31.25rem] h-[20rem] bg-gray-200 flex">
        <div className="w-14 bg-gray-400 h-full flex justify-center items-center">
          <div className="rotate-90 select-none">{RESERVATION}</div>
        </div>
        <div className="m-8 flex flex-col gap-2">
          <div>
            {RESERVATION_ID} <span className="font-bold">{reservationId}</span>
          </div>
          <div>
            {HALL} {hallNumber}
          </div>
          <div className="font-bold">{movie}</div>
          <div>{time}</div>

          <div>
            {SEATS}
            {seats.map((seat, index) => (
              <span key={`${seat.row}/${seat.column}`}>
                {seat.row}/{seat.column}
                {index !== seats.length - 1 && ", "}
              </span>
            ))}
          </div>
          <div>
            {NORMAL_TICKETS} {typesOfTickets.normal}
          </div>
          <div>
            {REDUCED_TICKETS} {typesOfTickets.reduced}
          </div>
          <div>
            {TOTAL_PRICE} {totalPrice} lei
          </div>
        </div>
      </div>
    </div>
  );
};

export default ReservationCard;
