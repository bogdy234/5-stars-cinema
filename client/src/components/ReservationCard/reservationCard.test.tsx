import renderer from "react-test-renderer";
import ReservationCard from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ReservationCard
        reservationId={"2"}
        hallNumber={0}
        movie={"batman"}
        time={""}
        seats={[]}
        typesOfTickets={{ normal: 2, reduced: 3 }}
        totalPrice={0}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
