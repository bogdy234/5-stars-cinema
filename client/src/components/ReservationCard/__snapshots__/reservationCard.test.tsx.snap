// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full"
>
  <div
    className="w-[18rem] sm:w-[25rem] md:w-[31.25rem] h-[20rem] bg-gray-200 flex"
  >
    <div
      className="w-14 bg-gray-400 h-full flex justify-center items-center"
    >
      <div
        className="rotate-90 select-none"
      >
        Rezervare
      </div>
    </div>
    <div
      className="m-2 md:m-8 flex flex-col gap-2"
    >
      <div>
        Rezervare 
         
        <span
          className="font-bold"
        >
          2
        </span>
      </div>
      <div>
        Sala
         
        0
      </div>
      <div
        className="font-bold"
      >
        batman
      </div>
      <div>
        
      </div>
      <div>
        Locuri: 
      </div>
      <div>
        Bilet intreg:
         
        2
      </div>
      <div>
        Bilet Copii/Elevi/Studenti:
         
        3
      </div>
      <div>
        Pret total:
         
        0
         lei
      </div>
    </div>
  </div>
</div>
`;
